composite/outlier-detection/src/source.ts

```diff
@@ -0,0 +1,48 @@
+import { Execute } from '@chainlink/types'
+import * as GenesisVolatility from '@chainlink/genesis-volatility-adapter'
+import * as XBTO from '@chainlink/xbto-adapter'
+
+export enum SourceDataProvider {
+  XBTO = 'xbto',
+  GenesisVolatility = 'genesisvolatility',
+}
+
+const isSourceDataProvider = (envVar?: string): envVar is SourceDataProvider =>
+  Object.values(SourceDataProvider).includes(envVar as any)
+
+export const getSourceDataProviders = (): SourceDataProvider[] => {
+  const sourceDataProviders = (process.env.SOURCE_ADAPTERS || '').split(',')
+  const adapters: SourceDataProvider[] = []
+
+  sourceDataProviders.forEach((provider) => {
+    if (isSourceDataProvider(provider)) adapters.push(provider as SourceDataProvider)
+  })
+
+  return adapters
+}
+
+export const getSourceImpl = (types: SourceDataProvider[]): Execute[] => {
```

Comment:
It's easier to use this type of mapping function if it maps just one item.

```suggestion
export const getSourceImpl = (type: SourceDataProvider): Execute => {
```

Then externally if I want to map N items:

```javascript
const adapters = types.map(getSourceImpl)
```

Or in our case, we could write it close to this:

```javascript
const sourceAdapters = config.sourceDataProviders.map(source.makeExecute)
```